app/models/bishop.rb

```diff
@@ -1,5 +1,21 @@
 class Bishop < Piece
 
+	def valid_move?(new_x, new_y)
+		position_exist?(new_x, new_y) && bishop_allowed_moves?(new_x, new_y) && (!(diagonally_obstructed?(new_x, new_y)))		
```

Comment:
I don't think we need to check is obstructed here right? we are checking that with is_obstructed method in piece model.